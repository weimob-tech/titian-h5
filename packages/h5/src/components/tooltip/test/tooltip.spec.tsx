import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiTooltip } from '../index';
import { TestContainer } from '../../common/test/container';

describe('渲染 ti-tooltip', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiTooltip],
      template: () => (
        <ti-tooltip content="提示文字提示文字提示文字">
          <span>按钮</span>
        </ti-tooltip>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('提示方向', async () => {
    const page = await newSpecPage({
      components: [TiTooltip, TestContainer],
      template: () => (
        <test-container>
          <ti-tooltip direction="top" content="提示文字提示文字提示文字">
            <span>按钮</span>
          </ti-tooltip>
          <ti-tooltip direction="bottom" content="提示文字提示文字提示文字">
            <span>按钮</span>
          </ti-tooltip>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('提示点击关闭', async () => {
    const page = await newSpecPage({
      components: [TiTooltip],
      template: () => (
        <ti-tooltip closeOnClick={true} content="提示文字提示文字提示文字">
          <span>按钮</span>
        </ti-tooltip>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
