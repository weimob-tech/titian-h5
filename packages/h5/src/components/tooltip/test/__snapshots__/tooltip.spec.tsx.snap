// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-tooltip 基本渲染 1`] = `
<ti-tooltip>
  <mock:shadow-root>
    <div class="titian-tooltip" part="">
      <slot></slot>
      <div class="titian-tooltip- titian-tooltip-content" part="" style="top: 0px; left: 0px; border-top-left-radius: 0;">
        <ti-svg-path-view class="titian-tooltip-arrow titian-tooltip-arrow-up titian-tooltip-half" use-mask="" viewbox="0 0 18 12" style="left: 0; top: 0.5px;"></ti-svg-path-view>
        <div class="titian-tooltip-inner" part="">
          提示文字提示文字提示文字
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <span>
    按钮
  </span>
</ti-tooltip>
`;

exports[`渲染 ti-tooltip 提示方向 1`] = `
<test-container>
  <!---->
  <ti-tooltip>
    <mock:shadow-root>
      <div class="titian-tooltip" part="">
        <slot></slot>
        <div class="titian-tooltip- titian-tooltip-content" part="" style="top: 0px; left: 0px; border-bottom-left-radius: 0;">
          <ti-svg-path-view class="titian-tooltip-arrow titian-tooltip-arrow-down titian-tooltip-half" use-mask="" viewbox="0 0 18 12" style="bottom: 0.5px; left: 0;"></ti-svg-path-view>
          <div class="titian-tooltip-inner" part="">
            提示文字提示文字提示文字
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <span>
      按钮
    </span>
  </ti-tooltip>
  <ti-tooltip>
    <mock:shadow-root>
      <div class="titian-tooltip" part="">
        <slot></slot>
        <div class="titian-tooltip- titian-tooltip-content" part="" style="top: 0px; left: 0px; border-top-left-radius: 0;">
          <ti-svg-path-view class="titian-tooltip-arrow titian-tooltip-arrow-up titian-tooltip-half" use-mask="" viewbox="0 0 18 12" style="left: 0; top: 0.5px;"></ti-svg-path-view>
          <div class="titian-tooltip-inner" part="">
            提示文字提示文字提示文字
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <span>
      按钮
    </span>
  </ti-tooltip>
</test-container>
`;

exports[`渲染 ti-tooltip 提示点击关闭 1`] = `
<ti-tooltip>
  <mock:shadow-root>
    <div class="titian-tooltip" part="">
      <slot></slot>
      <div class="titian-tooltip- titian-tooltip-content" part="" style="top: 0px; left: 0px; border-top-left-radius: 0;">
        <ti-svg-path-view class="titian-tooltip-arrow titian-tooltip-arrow-up titian-tooltip-half" use-mask="" viewbox="0 0 18 12" style="left: 0; top: 0.5px;"></ti-svg-path-view>
        <div aria-hidden="true" class="titian-tooltip-mask"></div>
        <div class="titian-tooltip-inner" part="">
          提示文字提示文字提示文字
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <span>
    按钮
  </span>
</ti-tooltip>
`;
